modules/querier/config.go

```diff
@@ -4,26 +4,27 @@ import (
 	"flag"
 	"time"
 
-	cortex_frontend "github.com/cortexproject/cortex/pkg/querier/frontend"
+	cortex_worker "github.com/cortexproject/cortex/pkg/querier/worker"
 	"github.com/cortexproject/cortex/pkg/util/grpcclient"
 )
 
 // Config for a querier.
 type Config struct {
-	QueryTimeout         time.Duration                `yaml:"query_timeout"`
-	ExtraQueryDelay      time.Duration                `yaml:"extra_query_delay,omitempty"`
-	MaxConcurrentQueries int                          `yaml:"max_concurrent_queries"`
-	Worker               cortex_frontend.WorkerConfig `yaml:"frontend_worker"`
+	QueryTimeout         time.Duration        `yaml:"query_timeout"`
+	ExtraQueryDelay      time.Duration        `yaml:"extra_query_delay,omitempty"`
+	MaxConcurrentQueries int                  `yaml:"max_concurrent_queries"`
+	Worker               cortex_worker.Config `yaml:"frontend_worker"`
 }
 
 // RegisterFlagsAndApplyDefaults register flags.
 func (cfg *Config) RegisterFlagsAndApplyDefaults(prefix string, f *flag.FlagSet) {
 	cfg.QueryTimeout = 10 * time.Second
 	cfg.ExtraQueryDelay = 0
 	cfg.MaxConcurrentQueries = 5
-	cfg.Worker = cortex_frontend.WorkerConfig{
-		MatchMaxConcurrency: true,
-		Parallelism:         2,
+	cfg.Worker = cortex_worker.Config{
+		MatchMaxConcurrency:   true,
+		MaxConcurrentRequests: cfg.MaxConcurrentQueries,
```

Comment:
Think we can remove this line because we're initialising it again here - https://github.com/grafana/tempo/pull/442/files#diff-a1fec687cd87c7167aee355f831ebae6beb368066e3bfbf092ef0aaf5e216aa2R94

And also, how does nested initialisation work for the same object? Seeing this for the first time..